### PR TITLE
Add store notices to the Cart and Checkout block templates

### LINKED ISSUE #42787: Notices in a session aren't being hydrated into checkout block on entry
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

If the user's session contains a notice, that notice isn't displayed on the checkout block or cart block page on page load. It is, however, shown in the `Store Notices` block if it exists on the page.

### Expected behavior

Store notices in the session are displayed when the block checkout or cart pages are loaded.

### Actual behavior

A store notice in the user session is not displayed when the block checkout or cart pages are loaded

### Steps to reproduce

```php
add_action('template_redirect', 'add_wc_notice_and_redirect_to_checkout');

function add_wc_notice_and_redirect_to_checkout() {
    if ( isset( $_GET['debug_notice'] ) ) {
        wc_add_notice( 'This is a debug notice!', 'error' );

        // Redirect to the checkout page
        wp_safe_redirect( wc_get_checkout_url() );
        exit;
    }
}
```

1. Add the code snippet above to your test site. It will add a notice and redirect you to the checkout page when the `debug_notice` URL parameter is present.
2. Visit `yoursite.com/?debug_notice`
3. You'll be redirected to the checkout page with no notice being displayed.

### WordPress Environment

-   WordPress version: 6.4.2
-   WooCommerce version: 8.3.1
-   WooCommerce Blocks version: 11.7.0 (Also present on 11.4.9)

-   OS: MacOS
-   Browser Brave
-   Version 1.60.125

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by tarunvijwani**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

If the user's session contains a notice, that notice wasn't displayed on the Checkout block or Cart block page on page load. It was happening because we removed `<!-- wp:woocommerce/store-notices /-->` block from the templates. 

In this PR, we're adding store notices back to Checkout block or Cart page templates. 

Closes #42787.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

```php
add_action('template_redirect', 'add_wc_notice_and_redirect_to_cart');

function add_wc_notice_and_redirect_to_cart() {
    if ( isset( $_GET['debug_notice'] ) ) {
        wc_add_notice( 'This is a debug notice!', 'error' );

        // Redirect to the cart page
        wp_safe_redirect( wc_get_cart_url() );
        exit;
    }
}
```

1. Add the code snippet above to your test site. It will add a notice and redirect you to the Cart page when the `debug_notice` URL parameter is present.
2. Visit `yoursite.com/?debug_notice`
3. You'll be redirected to the Cart page with a notice being displayed.

| Before | After |
|---|---|
|image|image|

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add store notices to the Cart and Checkout block templates. 
#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss [hunk 1]
```diff
@@ -150,3 +150,8 @@
 		@extend %info;
 	}
 }
+
+// To fix notice banner width for all themes.
+.woocommerce.wc-block-store-notices.alignwide {
+	max-width: var(--wp--style--global--wide-size);
+}
```
Comment (tarunvijwani) [inline plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss]: This is to fix the notice banner width for twenty-twenty-four and all other themes. 

| Before | After |
|---|---|
|image|image|

File: plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss [hunk 2]
```diff
@@ -150,3 +150,8 @@
 		@extend %info;
 	}
 }
+
+// To fix notice banner width for all themes.
```
Comment (nielslange) [inline plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss]:
```suggestion
// To fix notice banner width for all block themes.
```

The variable `--wp--style--global--wide-size` isn't available on classic themes, and this fix only works on block themes, as classic themes do not have the template files to which you applied the changes.

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @tarunvijwani. The change looks great. I left a minor suggestion, but I'm approving this PR, as my suggestion is not a blocker. Thanks again for fixing this. 🙌

**Comment by tarunvijwani** [comment]
@nielslange Thank you for reviewing the PR! 🙂

**Pull request merged**